### DISCUSSION TITLE
error processing encrypted ufdr file

### DISCUSSION THREAD

**Discussion by paulobreim** [sepinf-inc/IPED#1297] [Q&A]
When I try to index a file .ufdr I got the message bellow.
I tested with 4.03, 4.02, 4.01 and 3.18.15

java.lang.Exception: Error decoding datasource f:\myfile.ufdr
        at iped.engine.datasource.ItemProducer.run(ItemProducer.java:154) ~[iped-engine-4.0.1.jar:?]
Caused by: org.xml.sax.SAXParseException: Content is not allowed in prolog.
        at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source) ~[xercesImpl-2.12.2.jar:?]
        at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source) ~[xercesImpl-2.12.2.jar:?]
        at iped.engine.datasource.UfedXmlReader.read(UfedXmlReader.java:228) ~[iped-engine-4.0.1.jar:?]
        at iped.engine.datasource.UfedXmlReader.read(UfedXmlReader.java:207) ~[iped-engine-4.0.1.jar:?]
        at iped.engine.datasource.ItemProducer.run(ItemProducer.java:123) ~[iped-engine-4.0.1.jar:?]

Is it a new problem or I am doing some mistake?

tks
paulo

**Comment by lfcnassif** [comment]
What is the Physical Analyzer version used to generate the UFDR report? Could you share the full IPED processing log?

**Reply by paulobreim** [reply]
Good point!. I don't have that Physical Analyzer version at the moment, but I'll try to get it.
I think it has a good chance of being the version because I have 3 other UFDR that came from the same location and they all give this error.
I have 5 others that came from another location and had no problem indexing.
Opening this ufdr (as a zip) it appears to be fine.
For now follow here is LOG.
[log.txt](https://github.com/sepinf-inc/IPED/files/9466599/log.txt)

**Reply by lfcnassif** [reply]
Well, at first sight, the error suggests the report.xml file inside the ufdr (zip) root folder is corrupted, or at least not following the XML specification. I have already implemented some workaround years ago for a similar error to "fix" some report.xml malformed files. If CellebriteReader.exe can open the UFDR file, and if it is not too old (we don't have human resources to support old versions of third party software), I would need the report.xml file inside the UFDR to see if it is possible to implement another workaround...

**Reply by paulobreim** [reply]
Unfortunately I don't have the version of cellebrite that did the extraction.
The celebritte reader version is 2.48.1.3 (2021), and it opens the file normally.

**Reply by lfcnassif** [reply]
If you could share privately the report.xml file inside the UFDR, I can take a look next week.

**Reply by paulobreim** [reply]
sending

**Comment by bragg1** [comment]
Same problem. Cellebrite version: 7.57. IPED's latest version too.

Sent privately as well.

**Comment by lfcnassif** [comment]
Hi,

I just tested the report.xml files you both sent to me privately.

@paulobreim yours is completely corrupted and is not a valid xml:

![image](https://user-images.githubusercontent.com/7276994/189967558-0bd5bedb-ece2-4da5-807f-7d7cb3aa85ba.png)

@bragg1 yours worked fine here, with both 4.0.3 and 4.1-snapshot, maybe you sent me a different xml by mistake, because I saw the error in the processing log you've sent to me.

**Reply by paulobreim** [reply]
Thank you for the message.

Very interesting. I have about 8 or 9 UFDRs and they all have the report.xml like this, and they can all be read by CellebriteReader. So it's likely that these files were generated by some old version of Cellebrite that might encrypt the report.xml

**Reply by wladimirleite** [reply]
There is an explicit option to encrypt the UFDR, right?

**Reply by lfcnassif** [reply]
> There is an explicit option to encrypt the UFDR, right?

Hum... I didn't know about this. Without knowledge about the encryption scheme and where the key is stored (is it published by Cellebrite somewhere?), there is nothing we can do...

PS: I, at least, won't waste effort to reverse engineer the encryption.

**Reply by wladimirleite** [reply]
I will take a closer look to find out if there is anything we can do.

**Reply by paulobreim** [reply]
I'll try to index informing a folder that contains the UFDR instead of informing the file with full name. IPED must open this UFDR and index by looking at each file. Let's see what happens.

**Reply by lfcnassif** [reply]
> I'll try to index informing a folder that contains the UFDR instead of informing the file with full name. IPED must open this UFDR and index by looking at each file. Let's see what happens.

I don't recommend this as warned here: https://github.com/sepinf-inc/IPED/discussions/1272

**Reply by paulobreim** [reply]
good point.

**Reply by wladimirleite** [reply]
> There is an explicit option to encrypt the UFDR, right?

When generating the UFDR, it is possible to check the first option to "protect" the report, and additionally add a password.
Screenshot in Portuguese:
![image](https://user-images.githubusercontent.com/7217001/190441899-ef61bb37-e871-4ff3-bddb-feea488c76c9.png)

**Reply by paulobreim** [reply]
Curious that when you open it with cellebrite reader, it doesn't ask for any password and shows the UFDR normally.

**Reply by wladimirleite** [reply]
Comparing the report.xml from a test UFDR I created here, only the first option (protect, but no password) was used in the sample provided by @paulobreim.

**Reply by wladimirleite** [reply]
I took a closer look in a few "protected" UFDRs (first checkbox selected, but no user password defined).
1. All files (not only "report.xml") are encrypted. 
2. I tried to check if a simple cypher was used, but it doesn't seem to be the case. 
3. It looks like a cryptographic algorithm, with a fixed key used to generate and open these files using Cellebrite tools. 
4. Although it is possible to reverse engineer the encryption process, it doesn't seem something simple (at all).

**Reply by lfcnassif** [reply]
Thanks @tc-wleite

**Reply by pauloadilson** [reply]
Does it mean that is not possible to index UFDR files that are protected?
Thanks

**Reply by lfcnassif** [reply]
Yes, not with this tool.
